homeworks/src/homework4/part1/Triangle.java

```diff
@@ -0,0 +1,90 @@
+package homework4.part1;
+
+public class Triangle extends Figure {
+    private int side1;
+    private int side2;
+    private int side3;
+
+    public Triangle(){
+        setSide1(DEFAULT_SIDE);
+        setSide2(DEFAULT_SIDE);
+        setSide3(DEFAULT_SIDE);
+    }
+    public Triangle(int side1, int side2, int side3) {
+        if (side1 > 0) {
```

Comment:
You have in all figures the same logic - if (value<0) -> use some default, if not -> use value. This may be a static method in the interface.

this.side1=Figure.replaceNegativeByDefault(side1);
...